packages/opentelemetry-instrumentation-ws/src/index.ts

```diff
@@ -419,9 +423,10 @@ export class WSInstrumentation extends InstrumentationBase<WS> {
               parentSpan?.setAttributes({
                 [SemanticAttributes.HTTP_STATUS_CODE]: 101,
               });
-              parentSpan?.end();
-              self._requestSpans.delete(request);
```

Comment:
I don't think we want to delete the parent span anymore since we're not closing it but end span should be called by the initial outer function.

Reply:
I think we do still want to delete it here -- this is just cleaning up the in memory map of spans we have, deleting it from that map won't end it, it will just free up the memory so it can be GC'd later. The map is a `WeakMap` such that it might be ok to leave it in there indefinitely, but I think explicitly removing it is clearer and more robust